lib/primo/parameter_validatable.rb

```diff
@@ -23,20 +23,23 @@ module ParameterValidatable
     #######################################33
 
 
-
     def validate(params)
       validators.each do |validate|
         message = validate[:message][params]
         if !send(validate[:query], params)
-          raise error.new(message)
+          if params.respond_to?(:request) && params.request.respond_to?(:uri)
+            message += "\nEndpoint: #{params.request.uri.to_s}"
+          end
+
+          raise error_class.new(message)
```

Comment:
I think this should be moved to the specific validator where we do not need to check where the param is a request.

Reply:
@quoideneuf I made a PR against this branch with change that I'm proposing. Basically this method is supposed to be kept general and the specific stuff like what we're trying here is supposed to go to method that know for instance that `params.request.uri` exists (i.e. doesn't need to check with if)